packages/beacon-node/src/metrics/metrics/beacon.ts

```diff
@@ -71,6 +72,82 @@ export function createBeaconMetrics(register: RegistryMetricCreator) {
       help: "Current number of pending partial withdrawals",
     }),
 
+    eip7805: {
+      validInclusionListByteSize: register.counter({
+        name: "beacon_valid_inclusion_lists_size_bytes_total",
+        help: "Byte size of the inclusion list",
+      }),
+      invalidInclusionListByteSize: register.counter({
+        name: "beacon_invalid_inclusion_lists_size_bytes_total",
+        help: "Byte size of the inclusion list",
+      }),
+      inclusionListSeen: register.counter<{source: InclusionListSource}>({
+        name: "beacon_inclusion_lists_seen_total",
+        help: "Total number of seen inclusion lists (gossip, api)",
+        labelNames: ["source"],
+      }),
+      inclusionListValid: register.counter<{source: InclusionListSource}>({
+        name: "beacon_valid_inclusion_lists_total",
+        help: "Total number of valid inclusion lists",
+        labelNames: ["source"],
+      }),
+      inclusionListInvalid: register.counter<{source: InclusionListSource; reason: InvalidInclusionListReason}>({
+        name: "beacon_invalid_inclusion_lists_total",
+        help: "Total number of invalid inclusion lists",
+        labelNames: ["source", "reason"],
+      }),
+      inclusionListValidationTime: register.histogram<{source: InclusionListSource}>({
+        name: "beacon_inclusion_lists_validation_time_seconds",
+        help: "Time taken to validate inclusion list",
+        buckets: [0.01, 0.05, 0.1, 0.5, 1, 2, 5],
+        labelNames: ["source"],
+      }),
+      inclusionListProposed: register.counter({
+        name: "beacon_inclusion_lists_proposed_total",
+        help: "Total number of inclusion lists proposed",
+      }),
+      inclusionListBroadcasted: register.counter({
+        name: "beacon_inclusion_lists_broadcasted_total",
+        help: "Total number of inclusion lists proposed",
+      }),
+      inclusionListReceivedSecFromSlot: register.histogram({
+        name: "beacon_inclusion_lists_received_seconds_from_slot",
+        help: "Inclusion list arrival time since the clock slot",
+        buckets: [0, 2, 4, 6, 9, 10, 11, 12],
+      }),
+      inclusionListTransactionsSeen: register.counter<{source: InclusionListSource}>({
+        name: "beacon_inclusion_list_transactions_seen_total",
+        help: "Total number of transactions seen in inclusion lists",
+        labelNames: ["source"],
+      }),
+      inclusionListTransactionsIncluded: register.counter({
+        name: "beacon_inclusion_list_transactions_included_total",
+        help: "Total number of inclusion list transactions included into payload",
+      }),
```

Comment:
how can we measure this from CL side, we aren't sure if the EL includes any of the transactions in the payload, for us IL transactions are just opaque bytes